tests/Unit/Evolution/Initialization/Test_Evolution.cpp

```diff
@@ -325,6 +342,195 @@ void test_join() {
         step_number_within_slab_2, AdaptiveSteppingDiagnostics{7, 2, 40, 6, 13},
         slab_size_goal_2);
 }
+
+namespace history_projection {
+struct Metavariables {
+  static constexpr size_t volume_dim = 3;
+  struct system {
+    using variables_tag =
+        Tags::Variables<tmpl::list<TestHelpers::Tags::Vector<DataVector>>>;
+  };
+};
+
+static_assert(tt::assert_conforms_to_v<
+              Initialization::ProjectTimeStepperHistory<Metavariables>,
+              amr::protocols::Projector>);
+
+using variables_tag = Metavariables::system::variables_tag;
+using dt_variables_tag = db::add_tag_prefix<Tags::dt, variables_tag>;
+
+template <size_t Label>
+struct HistoryEntry : db::SimpleTag {
+  using type = variables_tag::type;
+};
+
+template <size_t Label>
+struct HistoryDeriv : db::SimpleTag {
+  using type = dt_variables_tag::type;
+};
+
+using ElementData =
+    tuples::TaggedTuple<domain::Tags::Element<3>, domain::Tags::Mesh<3>,
+                        Tags::HistoryEvolvedVariables<variables_tag>,
+                        HistoryEntry<0>, HistoryDeriv<0>, HistoryDeriv<1>>;
+
+ElementData element_data(gsl::not_null<std::mt19937*> gen,
+                         const ElementId<3>& element_id, const Mesh<3>& mesh,
+                         const TimeStepId& time_step_id0,
+                         const TimeStepId& time_step_id1) {
+  std::uniform_real_distribution<double> dist(-1.0, 1.0);
+  auto value0 = make_with_random_values<variables_tag::type>(
+      gen, make_not_null(&dist), mesh.number_of_grid_points());
+  auto deriv0 = make_with_random_values<dt_variables_tag::type>(
+      gen, make_not_null(&dist), mesh.number_of_grid_points());
+  auto deriv1 = make_with_random_values<dt_variables_tag::type>(
+      gen, make_not_null(&dist), mesh.number_of_grid_points());
+  TimeSteppers::History<variables_tag::type> history(4);
+  history.insert(time_step_id0, value0, deriv0);
+  history.insert(time_step_id1, decltype(history)::no_value, deriv1);
+  return {Element<3>(element_id, {}), mesh,
+          std::move(history),         std::move(value0),
+          std::move(deriv0),          std::move(deriv1)};
+}
+
+void test_p_project_history() {
```

Comment:
I was surprised there wasn't already a test for p-refinement; it appears I put the test into a separate file Test_TimeStepperHistory.  So either remove the separate file or move these tests there.